File: source/code/csharp/authentication/create_directory_github.cs
using Stormpath.SDK.Provider;

var githubDirectory = client.Instantiate<IDirectory>();
githubDirectory.SetName("My Github Directory");

await client.CreateDirectoryAsync(githubDirectory,
    options => options.ForProvider(
        client.Providers()
            .Github()
            .Builder()
            .SetClientId("YOUR_GITHUB_CLIENT_ID")
            .SetClientSecret("YOUR_GITHUB_CLIENT_SECRET")
            .Build()
        ));

File: source/code/java/authentication/create_directory_github.java
import com.stormpath.sdk.provider;
import com.stormpath.sdk.directory;
import com.stormpath.sdk.tenant;

Directory directory = client.instantiate(Directory.class);
directory.setName("My Github Directory");
directory.setDescription("A Github directory");

CreateDirectoryRequest request = Directories.newCreateRequestFor(directory).
            forProvider(Providers.GITHUB.builder()
                    .setClientId("YOUR_GITHUB_CLIENT_ID")
                    .setClientSecret("YOUR_GITHUB_CLIENT_SECRET")
                    .build()
            ).build();

Tenant tenant = client.getCurrentTenant();
directory = tenant.createDirectory(request);
